programs/bpf_loader/src/lib.rs

```diff
@@ -304,7 +305,7 @@ fn process_instruction_common(
     if program.is_executable() {
         debug_assert_eq!(
```

Comment:
I think this is actually a boolean expression, or at least easier to understand as such:
`debug_assert_eq!(first_instruction_account == 1, !(invoke_context.get_stack_height() > 1));`

Inverting both sides:
`debug_assert_eq!(first_instruction_account == 0, invoke_context.get_stack_height() > 1);`